controllers/links/user.js

```diff
@@ -1,26 +1,44 @@
 const {FirebaseDynamicLinks} = require('firebase-dynamic-links');
+const UsersFunction = require('../../databases/functions/users');
+const {User} = require('../../databases/models');
 
 module.exports = async (req, res) => {
-  const firebaseDynamicLinks = new FirebaseDynamicLinks(process.env.FIREBASE_API_KEY);
-  const firebaseDynamicLinkURL = process.env.FIREBASE_DYNAMIC_LINK_URL;
-  const firebaseDynamicLinkAndroidAppPackage =
-    process.env.FIREBASE_DYNAMIC_LINK_ANDROID_APP_PACKAGE;
-  const firebaseDynamicLinkIOSAppPackage = process.env.FIREBASE_DYNAMIC_LINK_IOS_APP_PACKAGE;
+  const {
+    BETTER_WEB_APP_URL,
+    FIREBASE_API_KEY,
+    FIREBASE_DYNAMIC_LINK_URL,
+    FIREBASE_DYNAMIC_LINK_ANDROID_APP_PACKAGE,
+    FIREBASE_DYNAMIC_LINK_IOS_APP_PACKAGE,
+    BETTER_APP_STORE_ID
+  } = process.env;
+
+  const firebaseDynamicLinks = new FirebaseDynamicLinks(FIREBASE_API_KEY);
+  const firebaseDynamicLinkURL = FIREBASE_DYNAMIC_LINK_URL;
+  const firebaseDynamicLinkAndroidAppPackage = FIREBASE_DYNAMIC_LINK_ANDROID_APP_PACKAGE;
+  const firebaseDynamicLinkIOSAppPackage = FIREBASE_DYNAMIC_LINK_IOS_APP_PACKAGE;
+
+  const {username} = req.params;
+
   try {
+    const targetUser = await UsersFunction.findUserByUsername(User, username);
+
+    const betterWebAppUrl = `${BETTER_WEB_APP_URL}/profile/${username}`;
+    const betterWebAppUrlWithId = `${betterWebAppUrl}?userId=${targetUser.user_id}`;
     const {shortLink} = await firebaseDynamicLinks.createLink({
-      dynamicLinkInfo: {
-        domainUriPrefix: `${firebaseDynamicLinkURL}`,
-        link: `${firebaseDynamicLinkURL}/users?username=${req.params.username}`,
-        androidInfo: {
-          androidPackageName: firebaseDynamicLinkAndroidAppPackage
-        },
-        iosInfo: {
-          iosBundleId: firebaseDynamicLinkIOSAppPackage
-        },
-        navigationInfo: {
-          enableForcedRedirect: true
-        }
-      }
+      longDynamicLink:
+        firebaseDynamicLinkURL +
+        '?link=' +
+        betterWebAppUrlWithId +
+        '&apn=' +
+        firebaseDynamicLinkAndroidAppPackage +
+        '&afl=' +
+        betterWebAppUrlWithId +
+        '&isi=' +
+        BETTER_APP_STORE_ID +
+        '&ibi=' +
+        firebaseDynamicLinkIOSAppPackage +
+        'ifl=' +
+        betterWebAppUrlWithId
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The string concatenation used to create the `longDynamicLink` is hard to read and maintain. It would be better to use a template literal for this purpose, which would make the code cleaner and easier to understand.
```diff
- firebaseDynamicLinkURL +
- '?link=' +
- betterWebAppUrlWithId +
- '&apn=' +
- firebaseDynamicLinkAndroidAppPackage +
- '&afl=' +
- betterWebAppUrlWithId +
- '&isi=' +
- BETTER_APP_STORE_ID +
- '&ibi=' +
- firebaseDynamicLinkIOSAppPackage +
- 'ifl=' +
- betterWebAppUrlWithId

+ `${firebaseDynamicLinkURL}?link=${betterWebAppUrlWithId}&apn=${firebaseDynamicLinkAndroidAppPackage}&afl=${betterWebAppUrlWithId}&isi=${BETTER_APP_STORE_ID}&ibi=${firebaseDynamicLinkIOSAppPackage}&ifl=${betterWebAppUrlWithId}`
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->